src/jenkins_job_insight/storage.py

```
@@ -549,6 +1521,297 @@ async def save_html_report(job_id: str, html_content: str) -> Path:
+) -> int:
+    """Set a classification for a test (e.g., FLAKY, REGRESSION).
+
+    Can be set by the AI during analysis or by humans.
+
+    Args:
+        job_id: Required — scopes the classification to a specific analysis job.
+        visible: Whether the classification is immediately visible.
+            Set to 0 during AI analysis; revealed after analysis completes.
+    """
+    if not job_id or not job_id.strip():
+        raise ValueError("job_id is required for test classification")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)